pallets/runtime/tests/src/relayer_test.rs

```diff
@@ -203,6 +202,56 @@ fn do_basic_relayer_paying_key_test() {
     );
 }
 
+#[test]
+fn update_polyx_limit_test() {
+    ExtBuilder::default()
+        .monied(true)
+        .build()
+        .execute_with(&do_update_polyx_limit_test);
+}
+fn do_update_polyx_limit_test() {
+    let bob = User::new(AccountKeyring::Bob);
+    let alice = User::new(AccountKeyring::Alice);
+
+    let set = |p: User, x| Relayer::update_polyx_limit(p.origin(), bob.acc(), x);
+    let add = |p: User, x| Relayer::increase_polyx_limit(p.origin(), bob.acc(), x);
+    let sub = |p: User, x| Relayer::decrease_polyx_limit(p.origin(), bob.acc(), x);
+
+    let test_update = |res, expected_res: Result<(), Error>, expected_limit| {
+        if let Err(e) = expected_res {
+            assert_noop!(res, e);
+        } else {
+            assert_ok!(res);
+        }
+        assert_subsidy(bob, Some((alice, expected_limit)));
+    };
+
+    let mut limit = 10u128;
+    setup_subsidy(bob, alice, limit);
+
+    // Bob tries to update his Polyx limit.  Not allowed
+    test_update(set(bob, 1_000_000u128), Err(Error::NotPayingKey), limit);
```

Comment:
Not sure why `u128` was kept on everything below. It shouldn't be necessary to type check the test and it just adds noise.